### DISCUSSION TITLE
How can you access the site root and html filename via variables?

### DISCUSSION THREAD

**Discussion by astroDimitrios** [quarto-dev/quarto-cli#10930] [Q&A] [answered]
### Description

I am overwriting the full html template for an extension I'm writing. Part of the webpage includes links to the current page. Is there a way to access the name of the html file quarto is creating to put in the template? Is there a variable for the site root?

**Comment by mcanouil** [comment]
Unclear what you want/need here without a concrete example.
There is `QUARTO_PROJECT_OUTPUT_FILES` (https://quarto.org/docs/projects/scripts.html#pre-and-post-render).

**Comment by astroDimitrios** [comment]
This is the [template line](https://github.com/astroDimitrios/carpentries-quarto/blob/8888f3a7d5783e60a6dd39384b4d7ca7094540f2/_extensions/carpentries/templates/title-block.html#L144), which is a link to the current page, I would like to replace the `{{this_page}}` reference with the html filename of the page being rendered if I can get it from Quarto.

**Reply by mcanouil** [reply, marked as answer]
What are those double brackets elements? Extensions are based on Pandoc template system which uses dollars.

Anyhow, take a look at Pandoc documentation: https://pandoc.org/MANUAL.html#partials especially the parts about variables such as "Variables set automatically".

**Reply by astroDimitrios** [reply]
Thanks for that link, I will try using `$pagetitle$` and play around with the sourcefile, outputfile options as well. The `{{}}` are mustache template variables that I haven't converted to Quarto / Pandoc yet :) (our build/docs system atm is based off [mustache templates](https://mustache.github.io/))
